src/plugins/intel_cpu/src/nodes/tensoriterator.cpp

```diff
@@ -293,7 +293,8 @@ MemoryPtr DynamicBuffer::create_buffer(const dnnl::engine& eng) {
     const auto estimated_iters = estimate_iters();
     const Shape _shape = Shape({count, static_cast<size_t>(abs_stride * estimated_iters), len/elem_size});
     auto _descCreator = BlockedDescCreator::getCommonCreators().at(LayoutType::ncsp);
-    auto new_buffer_desc = _descCreator->createSharedDesc(from->getDesc().getPrecision(), _shape);
+    auto prec = from->getDesc().getPrecision() == ov::element::f16 ? ov::element::f32 : from->getDesc().getPrecision();
```

Comment:
Why we need this change?
TensorIterator has nothing x64 or ARM specific from the impl perspective.

Reply:
this change depend on augru rnn layer if we don't set fp32, seg fault is created in loop after rnn layer

Reply:
Is doesn't mean we need to create such kind of WA.
TI should work correctly w/o dependency on parent operation. We need to analyze the issue and find real root-cause

Reply:
its became irrelevant change, dien.xml is worked without tensoriterator changes (all problem was depend on fp16 in rnn and calculate shapes in scheduler)